File: client/react/hrms/src/services/jobPostingStatusService.js
import axios from "axios";

export default class JobPostingStatusService {
  add(jobPostingStatus) {
    console.log(jobPostingStatus);
    return axios.post(
      "http://localhost:8080/api/jobpostingstatuses/add",
      jobPostingStatus
    );
  }

  getAll() {
    return axios.get("http://localhost:8080/api/jobpostingstatuses/getall");
  }

  getLastByJobPostingId(jobPostingId) {
    return axios.get(
      "http://localhost:8080/api/jobpostingstatuses/getlastbyjobpostingid?jobPostingId=" +
        jobPostingId
    );
  }
}

File: server/java/src/main/java/com/hrms/business/abstracts/JobPostingStatusService.java
package com.hrms.business.abstracts;

import com.hrms.core.utilities.results.DataResult;
import com.hrms.core.utilities.results.Result;
import com.hrms.entities.concretes.JobPostingStatus;

import java.util.List;

public interface JobPostingStatusService {
    Result add(JobPostingStatus jobPostingStatus);
    DataResult<JobPostingStatus> getLastByJobPostingId(int jobPostingId);
    DataResult<List<JobPostingStatus>> getAllByJobPostingId(int jobPostingId);
    DataResult<List<JobPostingStatus>> getAll();
}
